expr/interval_compound_literal.go

```diff
@@ -0,0 +1,192 @@
+package expr
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/substrait-io/substrait-go/proto"
+	"github.com/substrait-io/substrait-go/types"
+)
+
+// IntervalCompoundLiteral creates an interval compound literal
+type IntervalCompoundLiteral struct {
+	Years              int32
+	Months             int32
+	Days               int32
+	Seconds            int32
+	SubSeconds         int64
+	SubSecondPrecision types.TimePrecision
+	Nullability        types.Nullability
+}
+
+func (m IntervalCompoundLiteral) WithYears(years int32) IntervalCompoundLiteral {
+	m.Years = years
+	return m
+}
+
+func (m IntervalCompoundLiteral) WithMonths(months int32) IntervalCompoundLiteral {
+	m.Months = months
+	return m
+}
+
+func (m IntervalCompoundLiteral) WithDays(days int32) IntervalCompoundLiteral {
+	m.Days = days
+	return m
+}
+
+func (m IntervalCompoundLiteral) WithSeconds(seconds int32) IntervalCompoundLiteral {
+	m.Seconds = seconds
+	return m
+}
+
+func (m IntervalCompoundLiteral) WithMilliSecond(milliSeconds int64) IntervalCompoundLiteral {
+	m.SubSeconds = milliSeconds
+	m.SubSecondPrecision = types.PrecisionMilliSeconds
+	return m
+}
+
+func (m IntervalCompoundLiteral) WithMicroSecond(microSeconds int64) IntervalCompoundLiteral {
+	m.SubSeconds = microSeconds
+	m.SubSecondPrecision = types.PrecisionMicroSeconds
+	return m
+}
+
+func (m IntervalCompoundLiteral) WithNanoSecond(nanoSeconds int64) IntervalCompoundLiteral {
+	m.SubSeconds = nanoSeconds
+	m.SubSecondPrecision = types.PrecisionNanoSeconds
+	return m
+}
+
+func (m IntervalCompoundLiteral) WithSubSecond(subSeconds int64, precision types.TimePrecision) IntervalCompoundLiteral {
+	m.SubSeconds = subSeconds
+	m.SubSecondPrecision = precision
+	return m
+}
+
+func (m IntervalCompoundLiteral) WithNullability(nullability types.Nullability) IntervalCompoundLiteral {
+	m.Nullability = nullability
+	return m
+}
```

Comment:
since the members are all public now, do we still need these at all?

Reply:
yes, not useful now. removed them